test/drivers/test_rsvd.cc

```diff
@@ -17,91 +17,118 @@ class TestRSVD : public ::testing::Test
 
     virtual void TearDown() {};
 
-    /// General test for RSVD:
-    /// Computes the decomposition factors, then checks A-U\Sigma\transpose{V}.
-    template <typename T, typename RNG>
-    static void test_RSVD1_general(int64_t m, int64_t n, int64_t k, int64_t p, int64_t block_sz, T tol, std::tuple<int, T, bool> mat_type, RandBLAS::base::RNGState<RNG> state) {
-        
-        printf("|==================================TEST QB2 GENERAL BEGIN==================================|\n");
-
-        // For running QB
-        std::vector<T> A(m * n, 0.0);
-        RandLAPACK::util::gen_mat_type(m, n, A, k, state, mat_type);
-
-        int64_t size = m * n;
+    template <typename T>
+    struct RSVDTestData {
```

Comment:
Have this struct track parameters ``(m, n, k)``.

Reply:
Once again, d don't see the point exactly. 
This would also mess with readability, since we use those parameters a lot inside the functions.

Reply:
You can unpack the dimension parameters once you enter a function. A-la
```
auto m = testdata.m;
auto n = testdata.n;
auto k = testdata.k;
```

Reply:
Added.